public/framework/core_refs.ts

```diff
@@ -34,6 +35,7 @@ class CoreRefs {
   public dashboardProviders?: unknown;
   public overlays?: OverlayStart;
   public dataSource?: DataSourcePluginStart;
+  public dataSources?: DataSourceStart;
```

Comment:
@amsiglan Thanks for adding this. @sumukhswamy  are we using `dataSource` anywhere if not let's remove this.